### ISSUE TITLE
Column name beginning with number

### ISSUE THREAD

**Issue by dnnspaul** [kyleconroy/sqlc#1478]
Seems like sqlc doesn't handle columns correctly when it's starting with a number. Go structs can't have names starting with a number and that causes the error I think.

SQL schema:
```sql
CREATE TABLE IF NOT EXISTS `users` (
  `id` int unsigned NOT NULL AUTO_INCREMENT,
  `email` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci NOT NULL,
  `2fa_secret` tinyblob NOT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `UNIQUE email` (`email`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;
```

results in error:
![sqlc generate error](https://i.blaumedia.com/2022-03-07_22-40_NZs2gszgcaqPXsmcIXUwNvq9mdJiDkje.png)

the obvious fix is changing the columns name, but the error was very irritating for me and took me some time to find out the reason.
Maybe a simple "you can't use this here" would be very helpful.

**Comment by kyleconroy** [comment]
Yeah, there are a few ways we can solve this. First, the error message you're seeing is really unhelpful. We should error during generation with a clear message.

**Comment by DavidFluck** [comment]
Good afternoon. I'm poking around looking for ways to contribute, and in testing this, I noticed that sqlc now generates struct members which begin with an underscore for `2fa` in this example. Can we consider this resolved? cc @dnnspaul
